lib/util.js

```diff
@@ -420,6 +420,78 @@ exports.nextSleepTime = function (
     Math.min(base, previousSleep * 3));
 };
 
+
+/**
+ * Return next sleep time calculated by the jitter rule.
+ *
+ * @param {Number} numofRetries 
+ * @param {Number} currentSleepTime 
+ * @param {Number} maxSleepTime 
+ * @returns {JSON} return next sleep Time and totalTime.
+ */
+exports.jitteredSleepTime = function (numofRetries, currentSleepTime, maxSleepTime, totalTimeout = 0, maxLoginTimeout = 300) {
+  const nextSleepTime = Math.min(maxSleepTime, 2 ** numofRetries);
```

Comment:
The first option should be timeout remaining. The max sleep time should be decremented by the total timeout

Reply:
Fixed.